plugins/cuda/cuda_plugin.c

```diff
@@ -397,7 +460,7 @@ int cuda_plugin_pause_devices(int pid)
 		return -1;
 	}
 
-	if (add_pid_to_buf(&cuda_pids, pid)) {
+	if (add_pid_to_buf(&cuda_pids, restore_tid, CUDA_TASK_RUNNING)) {
```

Comment:
Shouldn't we be tracking the PID and not the restore thread id here?

Reply:
You're right, tracking the PID would indeed be more reliable here. Thank you for catching that!